lib/response-types/token-response-type.ts

```diff
@@ -0,0 +1,97 @@
+import { InvalidArgumentError } from '../errors';
+import { ImplicitGrantType } from '../grant-types';
+import { Client, Model, User } from '../interfaces';
+import { Request } from '../request';
+
+export class TokenResponseType {
+  accessToken: string;
+  accessTokenLifetime: number;
+  model: Model;
+  constructor(options: any = {}) {
+    if (!options.accessTokenLifetime) {
+      throw new InvalidArgumentError(
+        'Missing parameter: `accessTokenLifetime`',
+      );
+    }
+
+    this.accessToken = undefined;
+    this.accessTokenLifetime = options.accessTokenLifetime;
+    this.model = options.model;
+  }
+
+  /**
+   * Handle token response type.
+   */
+
+  async handle(
+    request: Request,
+    client: Client,
+    user: User,
+    uri: string,
+    scope: string,
+  ) {
+    if (!request) {
+      throw new InvalidArgumentError('Missing parameter: `request`');
+    }
+
+    if (!client) {
+      throw new InvalidArgumentError('Missing parameter: `client`');
+    }
+
+    const accessTokenLifetime = this.getAccessTokenLifetime(client);
+
+    const options = {
+      user,
+      scope,
+      model: this.model,
+      accessTokenLifetime,
+    };
+
+    const grantType = new ImplicitGrantType(options);
+    const token = await grantType.handle(request, client);
+    this.accessToken = token.accessToken;
+
+    return token;
+  }
+
+  /**
+   * Get access token lifetime.
+   */
+
+  getAccessTokenLifetime(client: Client) {
+    return client.accessTokenLifetime || this.accessTokenLifetime;
+  }
+
+  /**
+   * Build redirect uri.
+   */
+
+  buildRedirectUri(redirectUri: any) {
+    return this.setRedirectUriParam(
+      redirectUri,
+      'access_token',
+      this.accessToken,
+    );
+  }
+
+  /**
+   * Set redirect uri parameter.
+   */
+
+  setRedirectUriParam(redirectUri: any, key: string, value: any) {
+    if (!redirectUri) {
```

Comment:
As far as I can see `redirectUri` is always the result of `url.parse`, i.e. a "legacy urlObject" (https://nodejs.org/docs/latest/api/url.html#url_legacy_urlobject). This should be reflected in the type here, especially since the callers (`buildSuccessRedirectUri`/`buildErrorRedirectUri`) take also a `redirectUri` that is a `string`.